dot/rpc/http.go

```diff
@@ -56,13 +54,18 @@ type HTTPServerConfig struct {
 	WSEnabled           bool
 	WSPort              uint32
 	Modules             []string
-	WSSubscriptions     map[uint32]*WebSocketSubscription
 }
 
-// WebSocketSubscription holds subscription details
-type WebSocketSubscription struct {
-	WSConnection     *websocket.Conn
-	SubscriptionType int
+// WSConn struct to hold WebSocket Connection references
+type WSConn struct {
+	wsconn             *websocket.Conn
+	mu                 sync.Mutex
+	serverConfig       *HTTPServerConfig
+	logger             log.Logger
```

Comment:
might be nice to make the logger global to the rpc package

Reply:
good idea.